Add new weather data fetch script

smoskii opened the pull request:
## Summary

Added a new weather data fetch script that gets a CSV file of weather data from [Open-Meteo](https://open-meteo.com/en/docs?hourly=temperature_2m,apparent_temperature,precipitation_probability,precipitation,rain,showers,snowfall,snow_depth&daily=temperature_2m_max,temperature_2m_min,sunrise,sunset&current=temperature_2m,relative_humidity_2m,apparent_temperature,precipitation,rain,showers,snowfall,cloud_cover,wind_speed_10m&bounding_box=-90,-180,90,180&latitude=49.265&longitude=-123.242&timezone=America%2FLos_Angeles&models=best_match#location_and_time) 

## Changes

- [ ] Feature added - weather data fetch script that saves a CSV file containing raw weather data to data/raw/weather/ .
- [ ] Updated requirements.txt as needed
- [ ] Created weather folder and .gitkeep in data/raw


## Related Issues

`#7 Create Data Source for Weather Data`

## How to Test

When weather_fetch.py is run, it should save current weather data as a CSV file to data/raw/weather. The file should look like this:
<img width="1087" height="1050" alt="image" src="https://github.com/user-attachments/assets/bf1f3584-59fd-4866-96af-4096b495f3f8" />

## Notes

The data still needs to be cleaned and eventually incorporated into a workflow.

smoskii committed:
Add script to retrieve CSV file of weather data

smoskii committed:
Add dependencies to requirements.txt

smoskii committed:
Create .gitkeep for weather folder

smoskii committed:
Update download path for weather data CSV

smoskii committed:
Rename weather_fetch.py

smoskii committed:
Fixed error in weather_fetch.py

smoskii committed:
Fixed more code issues in weather_fetch.py

smoskii committed:
Fixed another small issue with weather_fetch.py

smoskii committed:
Fix weather_fetch.py (again...)

holiday-sean reviewed (changes requested):
Thanks for making the PR! There is just two quick things to resolve before merging for the weather_fetch script :

- <b>Line 4</b>: The download_url link has spaces (because of the triple quotes), so this messes up the API call (will receive a 400 Error). Perhaps try string concatenation with brackets? 

- <b>Line 15</b>: I think the file path should be changed to "../data/raw/weather/weather_data.csv". Because the 'data' folder is located outside the 'scripts' folder, the script currently cannot find where the weather data is.

smoskii committed:
Fixed download_url and file path

Used string concatenation for download_url and changed the file path to "../data/raw/weather/weather_data.csv

smoskii committed:
Fixed indendation for readability in download_url

smoskii committed:
Fixed indentation and deleted blank lines at end of file

holiday-sean reviewed (approved):
Thank you for resolving the issues!

kashish1928 reviewed (approved):
LGTM !
